### PR TITLE
ice 3.6.1

### PR DESCRIPTION
**Pull request by externl**
Update to Ice 3.6.1 and fix Java to require at least java 7 (6 does not work.) and Berkeley DB to build Java 7 compatible code.

### CODE REVIEW

File: Library/Formula/ice.rb [hunk 1]
```diff
@@ -32,7 +32,10 @@ def install
         --enable-cxx
       ]
 
-      args << "--enable-java" if build.with? "java"
+      if build.with? "java"
+        args << "--enable-java"
+        inreplace "dist/Makefile.in", "@JAVACFLAGS@",  "@JAVACFLAGS@ -source 1.7 -target 1.7"
```
Comment (dunn) [inline Library/Formula/ice.rb]: Should this be submitted as a patch upstream?

Comment (externl) [inline Library/Formula/ice.rb]: No, Oracle is not doing anything with DB 5 anymore.

Comment (mikemcquaid) [inline Library/Formula/ice.rb]: It should still be reported.

Comment (externl) [inline Library/Formula/ice.rb]: TBO I'm not really sure it's an issue with the Berkeley DB build system. The distros which ship it just build with Java 7.

The issue is that if I have Java 7 and 8 installed on my system and I want to build a formula, there's no way to specify a version (unless I missed it in the docs). Setting JAVA_HOME does not work. Brew will always choose Java 8.

Comment (xu-cheng) [inline Library/Formula/ice.rb]: > Brew will always choose Java 8

`depends_on :java => "1.7"`

Comment (externl) [inline Library/Formula/ice.rb]: It does not depend on Java 1.7 though, Java 1.8 builds fine. The issue is if I build with Java 8 (the default if both are installed) then want to use it with Java 7 only project. Which is what this replace fixes.

Comment (externl) [inline Library/Formula/ice.rb]: I will still report this to Oracle. That aside is everyone okay with this in the meantime?

**Comment by externl** [comment]
As a side node I have another commit I was going to make which fixes the build on El Capitan. Would like to get this one in first though.

File: Library/Formula/ice.rb (hunk 1 above)
Comment (externl) [inline Library/Formula/ice.rb]: Anyone?

Comment (DomT4) [inline Library/Formula/ice.rb]: Can you include a link to that bug submission in the formula? If we can get that done, I'm happy at least.

Comment (externl) [inline Library/Formula/ice.rb]: It's an internal bug submission system, there is no public access. Also, the version of db used by Ice comes from ZeroC, which already includes various patches. I don't see why this one line patch is such a big deal? Especially considering Oracle is no longer supported 5.3 and this version is specific just for ZeroC Ice.

Comment (externl) [inline Library/Formula/ice.rb]: Suggestions? I can just make it a part of our Berkeley DB zip?

Comment (externl) [inline Library/Formula/ice.rb]: Would love to get this in so I can submit an El Capitan fix! :)

Comment (dunn) [inline Library/Formula/ice.rb]: Can you add a comment about having submitted a ticket upstream, so it doesn't end up as a floating patch?

Comment (externl) [inline Library/Formula/ice.rb]: Will do.

**Comment by externl** [comment]
I've added the comment for Berkeley DB. I've also added the various things needed for El Capitan support.

**Comment by dunn** [comment]
Thanks for your patience here; merged in e81b5b03bdb0045c9589093afbaf09136a208d28 ❄️

**Pull request closed without merging**